src/test/java/io/getunleash/repository/FeatureToggleRepositoryTest.java

```diff
@@ -156,6 +157,27 @@ public void should_perform_synchronous_fetch_on_initialisation() {
         verify(toggleFetcher, times(1)).fetchToggles();
     }
 
+    @Test
+    public void should_perform_synchronous_fetch_on_initialisation_and_fail_if_invalid_upstream() {
```

Comment:
Can we add a test for the opposite as well, that we should not throw if` `synchronousFetchOnInitialisation=false`

Reply:
We have many tests with this configuration, but I believe we have no test validating that after the initial failure the next interval we're going to try again. Let me check and add that

Reply:
Added here: [fef2827](https://github.com/Unleash/unleash-client-java/pull/196/commits/fef2827de4bad7290ba77c7ee06f660b12916987) and checked with @chriswk 
I'll proceed with the merge